### PR TITLE
Eventonica 

### PR DESCRIPTION
**Pull request by priyaraj7**
Hi Chris, I have updated the eventonica doc. Could you please review it. Thank you

### CODE REVIEW

File: project-example/week-9/eventonica/eventonica-express-backend.md [hunk 1]
```diff
@@ -20,116 +20,90 @@ So let's get to it!
...
+   npm install express cors
+   npm install --save-dev nodemon
    ```
 
-1. Stop your `server` app and restart. `http://localhost:4000` should now show your new message.
+- Let’s take a quick look at the three packages:
 
-1. `.gitignore` your `node_modules`. Push your project up to GitHub.
+  - express: Express is a fast and lightweight web framework for Node.js. Express is an essential part of the PERN stack.
+  - cors: CORS is a node.js package for providing an Express middleware that can be used to enable CORS with various options. Cross-origin resource sharing (CORS) is a mechanism that allows restricted resources on a web page to be requested from another domain outside the domain from which the first resource was served.
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-express-backend.md]: Nice addition! It could also use an explanation of why we need it in our case. E.g. We need cors because we're running the client and server apps on different domains (because the ports are different). If they were running on the same domain, we wouldn't need to use this.

File: project-example/week-9/eventonica/eventonica-express-backend.md [hunk 2]
```diff
@@ -20,116 +20,90 @@ So let's get to it!
...
+  - nodemon: nodemon is a tool that helps develop Node.js based applications by automatically restarting the node application when file changes in the directory are detected.
 
-#### Convert CJS module to EcmaScript Modules(ESM)
+- Next, go into the package.json `scripts` and add this start script. This help you to run server with a simple command. Tell Node.js that all files are ES Modules by adding `"type": "module"`
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-express-backend.md]: This section starts out as a numbered list, but then switches to a bullet list. Probably should switch this back to a number:
```suggestion
1. Next, go into the package.json `scripts` and add this start script. This helps you to run the server with a simple command. Tell Node.js that all files are ES Modules by adding `"type": "module"`
```

File: project-example/week-9/eventonica/eventonica-express-backend.md [hunk 3]
```diff
@@ -139,32 +113,73 @@ The example response returns `{ users: [array of users] }` instead of just `res.
...
 
-To test this, you can console log `console.log(req.body, 'the body')` before the `res.json` line. In Postman, now try adding a body. Click the "body" tab and select "raw". Then select "JSON" from the dropdown. Try sending a JSON of something that the API might send. For example, it could send `{"name": "nemo"}` for when you implement filtering. If you added the console log, do you see this information printed in your express server terminal? (Note that console logs will not show up in Postman).
+To test this, you can console log `console.log(req.body, 'the body')` before the `res.json` line. In Postman, now try adding a body. Click the "body" tab and select "raw". Then select "JSON" from the dropdown. Try sending a JSON of something that the API might send. For example, it could send `{"name": "nemo"}` for when you implement filtering.
+Right now you can see `undefined` in console. Add Express body parser `app.use(express.json());` in index.js. Now you can see `{"name": "nemo"}` in console.
+Note: that console logs will not show up in Postman.
+
+Your final code look something like this:
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-express-backend.md]:
```suggestion
Your final code should look something like this:
```

File: project-example/week-9/eventonica/eventonica-express-backend.md [hunk 4]
```diff
@@ -176,37 +191,25 @@ To test this, you can console log `console.log(req.body, 'the body')` before the
...
 
 Add remaining REST API routes for `users` listed in the [project README](./README.md). Don't worry about `events` or `favorites` for now... you can add those after we get our database going.
 
-`Users` now needs "add" and "delete" functionality. For example, a frontend function called `addUser()` should make a POST request to http://localhost:4000/users/
+`Users` now needs "add" and "delete" functionality. For example, a frontend function called `addUser()` should make a POST request to http://localhost:4000/api/users/
 and add a user by posting JSON with your API (which currently just saves to a variable since we have no DB), and the API would need a route like this:
 
 ```js
-router.post('/', function (req, res, next) {
+app.post('/', function (req, res, next) {
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-express-backend.md]:
```suggestion
app.post('/api/users', function (req, res, next) {
```

File: project-example/week-9/eventonica/eventonica-postgres.md [hunk 5]
```diff
@@ -214,14 +248,68 @@ const handleSubmit = async (e) => {
...
 ```
 
-### Add Remaining Functions
+```js
+// Edit user
 
-- Add code for delete users.
+const editUser = (user) => {
+  setIsEditingUser(true);
+  setCurrentUser({ ...user });
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-postgres.md]:
```suggestion
  setCurrentUser(user);
```

File: project-example/week-9/eventonica/eventonica-react-ui.md [hunk 6]
```diff
@@ -119,112 +195,347 @@ document
...
 ```jsx
 // id, name, and email are states that store what values the user types in those fields
 // users is an array of user objects
 // All of these states can be defined in the component
-const handleSubmit = (e) => {
+const handleAddSubmit = (e) => {
+  // Prevent browser refreshing after form submission
   e.preventDefault();
-  const newUser = { id: id, name: name, email: email };
+  const newUser = { id: values.id, name: values.name, email: values.email };
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-react-ui.md]:
```suggestion
  const newUser = { id: inputValues.id, name: inputValues.name, email: inputValues.email };
```

File: project-example/week-9/eventonica/eventonica-react-ui.md [hunk 7]
```diff
@@ -119,112 +195,347 @@ document
...
+  const value = e.target.value;
+  setValues({
+    ...inputValues,
+    [e.target.name]: value
+  });
+};
+```
 
-1. Update the HTML form under "Add User" to have an ID field and an email field.
+Open your application in a web browser. Enter user name, email and id. click the submit button. You will see the user data in table
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-react-ui.md]:
```suggestion
Open your application in a web browser. Enter user name, email and id. Click the submit button. You will see the user data in table.
```

File: project-example/week-9/eventonica/eventonica-sample-code/client/src/components/Users.jsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+function Users() {
+  const [users, setUsers] = useState([]);
+
+  const values = { name: "", email: "", id: "" };
+
+  // boolean state to know if we are editing (this will let us display
+  // different inputs based on a condition (conditional rendering)
+  const [isEditingUser, setIsEditingUser] = useState(false);
+  // object state to set so we know which userwe are editing
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-sample-code/client/src/components/Users.jsx]:
```suggestion
  // object state to set so we know which user we are editing
```

File: project-example/week-9/eventonica/eventonica-sample-code/client/src/components/Users.jsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+    setIsEditingUser(false);
+  };
+  return (
+    <section className="user-management">
+      <h2>User Management</h2>
+
+      <UserList users={users} editUser={editUser} deleteUser={deleteUser} />
+
+      {/* FOrm functionality */}
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-sample-code/client/src/components/Users.jsx]:
```suggestion
      {/* Form functionality */}
```

File: project-example/week-9/eventonica/eventonica-postgres.md [hunk 10]
```diff
@@ -65,94 +65,112 @@ In addition to the usual steps:
...
        return res.status(400).json({ e });
      }
    });
 
-   /* post request goes here */
-
-   /* delete request goes here  */
+      <!-- POST request goes here -->
+      <!-- PUT request goes here -->
+      <!-- DELETE request goes here -->
```
Comment (chrisjaure) [inline project-example/week-9/eventonica/eventonica-postgres.md]: Might want to switch these back to javascript comments rather than html comments.

**Review by chrisjaure — COMMENTED** [review]
This is a lot to review! I have to admit I started skimming it after a while 😅 

Could you summarize your changes in the PR description or point out anything you want reviewed more thoroughly please?

**Comment by priyaraj7** [comment]
> This is a lot to review! I have to admit I started skimming it after a while 😅
> 
> Could you summarize your changes in the PR description or point out anything you want reviewed more thoroughly please?

Thank you so much for reviewing this project. In future, I will summarize the PR and create small PR. I appreciate your time and guidance.
